db/schema_applier.hh

```diff
@@ -22,6 +23,71 @@ namespace schema_tables {
 
 future<> merge_schema(sharded<db::system_keyspace>& sys_ks, distributed<service::storage_proxy>& proxy, gms::feature_service& feat, std::vector<mutation> mutations, bool reload = false);
 
+enum class table_kind { table, view };
+
+struct table_selector {
+    bool all_in_keyspace = false; // If true, selects all existing tables in a keyspace plus what's in "tables";
+    std::unordered_map<table_kind, std::unordered_set<sstring>> tables;
+
+    table_selector& operator+=(table_selector&& o);
+    void add(table_kind t, sstring name);
+    void add(sstring name);
+};
+
+struct schema_persisted_state {
+    schema_tables::schema_result keyspaces;
+    schema_tables::schema_result scylla_keyspaces;
+    std::map<table_id, schema_mutations> tables;
+    schema_tables::schema_result types;
+    std::map<table_id, schema_mutations> views;
+    schema_tables::schema_result functions;
+    schema_tables::schema_result aggregates;
+    schema_tables::schema_result scylla_aggregates;
+};
+
+// Schema_applier encapsulates intermediate state needed to construct schema objects from
+// set of rows read from system tables (see struct schema_state). It does atomic (per shard)
+// application of a new schema.
+class schema_applier {
+    using keyspace_name = sstring;
+
+    sharded<service::storage_proxy>& _proxy;
+    sharded<db::system_keyspace>& _sys_ks;
+    const bool _reload;
+
+    std::set<sstring> _keyspaces;
+    std::unordered_map<keyspace_name, table_selector> _affected_tables;
+    locator::tablet_metadata_change_hint _tablet_hint;
+
+    schema_persisted_state _before;
+    schema_persisted_state _after;
+
+    future<schema_persisted_state> get_schema_persisted_state();
+public:
+    schema_applier(
+            sharded<service::storage_proxy>& proxy,
+            sharded<db::system_keyspace>& sys_ks,
+            bool reload = false)
```

Comment:
What is reload?

Reply:
the it some legacy api (I think http based) to  force reloading the schema, I kept it working.